frontend/src/components/core/Sidebar/SidebarNav.tsx

```diff
@@ -47,121 +54,120 @@ export interface Props {
 const SidebarNav = ({
   appPages,
   collapseSidebar,
+  collapseNav,
   currentPageScriptHash,
   hasSidebarElements,
-  hideParentScrollbar,
   onPageChange,
   pageLinkBaseUrl,
 }: Props): ReactElement | null => {
-  if (appPages.length < 2) {
-    return null
-  }
-
-  const [expanded, setExpanded] = useState(false)
+  const [expanded, setExpanded] = useState(!collapseNav)
+  const [pagesToShow, setPagesToShow] = useState(
+    collapseNav === true ? 6 : appPages.length
+  )
   const navItemsRef = useRef<HTMLUListElement>(null)
-  const isOverflowing = useIsOverflowing(navItemsRef)
+  const toggleButtonRef = useRef<HTMLButtonElement>(null)
   // We use React.useContext here instead of destructuring it in the imports
   // above so that we can mock it in tests.
   const { getBaseUriParts } = React.useContext(AppContext)
 
-  const onMouseOver = useCallback(() => {
-    if (isOverflowing) {
-      hideParentScrollbar(true)
-    }
-  }, [isOverflowing, hideParentScrollbar])
-
-  const onMouseOut = useCallback(
-    () => hideParentScrollbar(false),
-    [hideParentScrollbar]
+  const toggleExpanded = useCallback(
+    (target = null) => {
+      if (!expanded) {
+        setExpanded(true)
+        setPagesToShow(appPages.length)
+
+        if (target === toggleButtonRef.current && localStorageAvailable()) {
+          localStorage.setItem("navExpanded", "true")
+        }
+      } else {
+        setExpanded(false)
+        setPagesToShow(6)
+
+        if (target === toggleButtonRef.current && localStorageAvailable()) {
+          localStorage.setItem("navExpanded", "false")
+        }
+      }
+    },
+    [expanded]
   )
 
-  const toggleExpanded = useCallback(() => {
-    if (!expanded && isOverflowing) {
-      setExpanded(true)
-    } else if (expanded) {
-      setExpanded(false)
+  useEffect(() => {
+    if (collapseNav !== !expanded) {
```

Comment:
Nit: Prefer to avoid a double negative for readability's sake, perhaps `collapseNav === expanded`?

Reply:
Makes sense! Fixed on [773fa1b](https://github.com/streamlit/streamlit/pull/6050/commits/773fa1bcf187fc49d7310c3fd17980db2afdb448)